src/IOCapture.jl

```diff
@@ -49,11 +50,16 @@ It is also possible to set `throwerrors = :interrupt`, which will make `iocaptur
 only `InterruptException`s. This is useful when you want to capture all the exceptions, but
 allow the user to interrupt the running code with `Ctrl+C`.
 """
-function iocapture(f; throwerrors::Union{Bool,Symbol}=true)
-    # Currently, :interrupt is the only valid Symbol value for throwerrors
-    if isa(throwerrors, Symbol) && throwerrors !== :interrupt
-        throw(DomainError(throwerrors, "Invalid value passed for throwerrors"))
-    end
+function iocapture(f; throwerrors=Any)
+    # `throwerrors` is set to one of `true`, `false`, `:interrupt`, or a
+    # subtype of `ErrorException`, or a `Union` of error subtypes. Here we
+    # convert the first three choices to types instead, as:
+    #
+    #   - `true` -> `Any`,
+    #   - `false` -> `Union{}`,
+    #   - `:interrupt` -> `InterruptException`.
+    throwerrors = rewrite_error_argument(throwerrors)
```

Comment:
Ain't terribly type-stable, but considering what else usually gets called in `iocapture` I doubt it's of any concern.